lib/features/composer/presentation/manager/composer_manager.dart

```diff
@@ -1,35 +1,280 @@
 
   void addComposer(ComposerArguments composerArguments) {
-    final String id = DateTime.now().millisecondsSinceEpoch.toString();
-    log('ComposerManager::addComposer:ID = $id');
-    ComposerBindings(
-      composerId: id,
-      composerArguments: composerArguments,
-    ).dependencies();
-    composers[id] = Get.find<ComposerController>(tag: id);
-    log('ComposerManager::addComposer:Success');
+    final id = DateTime.now().millisecondsSinceEpoch.toString();
+    ComposerBindings(composerId: id, composerArguments: composerArguments).dependencies();
+
+    composers[id] = ComposerView(key: Key(id), composerId: id);
+    composerIdsQueue.add(id);
+
+    _syncQueueIfNeeded();
   }
 
   void removeComposer(String id) {
-    log('ComposerManager::removeComposer:ID = $id');
-    if (composers.containsKey(id)) {
-      ComposerBindings(composerId: id).dispose();
-      composers.remove(id);
-      log('ComposerManager::removeComposer:Success');
+    if (!composers.containsKey(id)) return;
+
+    composerIdsQueue.remove(id);
+    composers.remove(id);
+    ComposerBindings(composerId: id).dispose();
+
+    _syncQueueIfNeeded();
+  }
+
+  void _syncQueueIfNeeded() {
+    if (currentContext != null && composerIdsQueue.isNotEmpty) {
+      syncComposerStateWhenComposerQueueChanged(
+        screenWidth: _responsiveUtils.getSizeScreenWidth(currentContext!),
+      );
     }
   }
 
-  bool get hasComposer => composers.isNotEmpty;
+  ({
+    String? previousTargetId,
+    String? targetId,
+    String? nextTargetId,
+  }) findSurroundingComposerIds(String targetId) {
+    var ids = composerIdsQueue.toList(),
+        index = ids.indexOf(targetId);
+
+    return index == -1
+      ? (previousTargetId: null, targetId: null, nextTargetId: null)
+      : (
+          previousTargetId: index > 0 ? ids[index - 1] : null,
+          targetId: ids[index],
+          nextTargetId: index < ids.length - 1 ? ids[index + 1] : null
+        );
+  }
+
+  String? findFirstNormalComposerIdInQueue() {
+    return composerIdsQueue
+      .toList()
+      .reversed
+      .firstWhereOrNull((id) => getComposerView(id).controller.isNormalScreen);
+  }
+
+  (
+    double availableWidth,
+    double totalWidth,
+    Map<String, ComposerController> composerControllers
+  ) _setUpDataToSyncComposerQueue({required double screenWidth}) {
+    double totalWidth = 0;
+    double countDisplayedComposer = 0;
+    final composerControllers = <String, ComposerController>{};
+
+    for (var id in composerIdsQueue) {
+      final controller = getComposerView(id).controller;
+      final composerWidth = controller.composerWidth;
+      if (!controller.isHiddenScreen) {
+        countDisplayedComposer++;
+      }
+      composerControllers[id] = controller;
+      totalWidth += composerWidth;
+    }
+
+    final availableWidth = screenWidth
+      - ComposerStyle.composerExpandMoreButtonMaxWidth
+      - ComposerStyle.padding * 3
```

Comment:
Why need `x * 3`? can you constant it?

Reply:
Oh, it should be `x * 2`. that is horizontal padding. We just need to const the padding value on one side.

Reply:
Update to `ComposerStyle.padding * 2`